pymeasure/instruments/keysight/keysightDSOX1102G.py

```diff
@@ -0,0 +1,559 @@
+    )
+
+    probe_attenuation = Instrument.control(
+        "PROBe?", "PROBe %f",
+        """ A float parameter that specifies the probe attenuation. The probe attenuation
+        may be from 0.1 to 10000.""",
+        validator=strict_range,
+        values=[0.1, 10000]
+    )
+
+    range = Instrument.control(
+        "RANGe?", "RANGe %f",
+        """ A float parameter that specifies the full-scale vertical axis in Volts. 
+        When using 1:1 probe attenuation, legal values for the range are from 8 mV to 40V."""
+    )
+
+    scale = Instrument.control(
+        "SCALe?", "SCALe %f",
+        """ A float parameter that specifies the vertical scale, or units per division, in Volts."""
+    )
+
+    def __init__(self, instrument, number):
+        self.instrument = instrument
+        self.number = number
+
+    def values(self, command, **kwargs):
+        """ Reads a set of values from the instrument through the adapter,
+        passing on any key-word arguments.
+        """
+        return self.instrument.values(":channel%d:%s" % (
+            self.number, command), **kwargs)
+
+    def ask(self, command):
+        self.instrument.ask(":channel%d:%s" % (self.number, command))
+
+    def write(self, command):
+        self.instrument.write(":channel%d:%s" % (self.number, command))
+
+    def setup(self, bwlimit: bool = None, coupling: str = None,
+                    display: bool = None, invert: bool = None, label: str = None,
+                    offset: float = None, probe_attenuation: float = None,
+                    vertical_range: float = None, scale=None):
+        """ Setup channel. Unspecified settings are not modified. Modifying values such as
+        probe attenuation will modify offset, range, etc. Refer to oscilloscope documentation and make
+        multiple consecutive calls to setup() if needed.
+
+        :param bwlimit: Enable 25 MHz internal low-pass filter.
+        :param coupling: "ac" or "dc".
+        :param display: Enable channel display.
+        :param invert: Enable input signal inversion.
+        :param label: Label string with max. 10 characters, may contain commonly used ASCII characters.
+        :param offset: Value represented at center of screen, must be inside the legal range.
+        :param probe_attenuation: Probe attenuation values from 0.1 to 1000.
+        :param vertical_range: Full-scale vertical axis of the selected channel. When using 1:1 probe
+                                attenuation, legal values for the range are  from 8mV to 40 V. If the
+                                probe attenuation is changed, the range value is multiplied by the
+                                probe attenuation factor.
+        :param scale: Units per division. """
+
+        if vertical_range is not None and scale is not None:
+            raise Warning('Both "vertical_range" and "scale" are specified. Specified "scale" has priority.')
```

Comment:
Any thoughts over the use of warnings.warn() vs log.warning()?

Here's the logging documentation (https://docs.python.org/3/howto/logging.html):
_warnings.warn() in library code if the issue is avoidable and the client application should be modified to eliminate the warning

logging.warning() if there is nothing the client application can do about the situation, but the event should still be noted_

Since this case is more of a "good to know" that a library-breaking issue, I'd favor your suggestion and replace the raise Warning() by log.warning().

Reply:
Hm, tanks, I was not aware of that distinction. I was more going after usage in this codebase - it always uses log.warning(), never Warning().